viewer/db.js

```diff
@@ -987,31 +1003,273 @@ exports.getHunt = async (id) => {
+// shortcut names by appending the cluster or index name to the shortcut name
+async function initialShortcutsSyncToRemote () {
+  if (internals.multiES) { return; } // don't sync shortcuts for multies
+
+  const { body: doc } = await internals.usersClient7.indices.getMapping({
+    index: internals.remoteShortcutsIndex
+  });
+
+  let initSync;
+  // eslint-disable-next-line no-unreachable-loop
+  for (const index in doc) {
+    initSync = doc[index]?.mappings?._meta?.initSync;
+    break;
+  }
+
+  if (initSync) { return; } // already been synced, don't need to do anything
+
+  const msg = `initial sync from local shortcuts index (${internals.localShortcutsIndex}) to remote index (${internals.remoteShortcutsIndex})`;
+
+  console.log(msg);
+
+  const compareArrays = (a, b) => {
+    if (!a && b) return false;
+    if (!b && a) return false;
+    if (!a && !b) return true;
+    if (a.length !== b.length) return false;
+    const uniqueValues = new Set([...a, ...b]);
+    for (const v of uniqueValues) {
+      const aCount = a.filter(e => e === v).length;
+      const bCount = b.filter(e => e === v).length;
+      if (aCount !== bCount) return false;
+    }
+    return true;
+  };
+
+  try {
+    const dbOperations = [];
+    // fetch shortcuts from remote and local indexes
+    const [{ body: remoteResults }, { body: localResults }] = await Promise.all([
+      exports.searchShortcuts({ size: 10000 }), exports.searchShortcutsLocal({ size: 10000 })
+    ]);
+
+    if (!localResults.hits.hits.length) { return; } // nothing to sync to remote db
+
+    // add each of the local shortcuts to the remote shortcuts db (remote db = user's es)
+    // but first, compare the local shortcuts to the remote shortcuts to determine
+    // if any shortcuts have the same name in the remote db (we can assume that
+    // none of the local shortcuts will be the same as the remote db shortcuts
+    // since they have not been synced before)
+    for (const localShortcut of localResults.hits.hits) {
+      let alreadyIndexed = false;
+      for (const remoteShortcut of remoteResults.hits.hits) {
+        if (remoteShortcut._source.name === localShortcut._source.name) {
+          // found a local shortcut with the same name as the remote
+          let diff = false; // compare the shortcuts values
+          if (localShortcut._source.shared !== remoteShortcut._source.shared) { diff = true; }
+          if (localShortcut._source.userId !== remoteShortcut._source.userId) { diff = true; }
+          if (localShortcut._source.description !== remoteShortcut._source.description) { diff = true; }
+          if (!compareArrays(localShortcut._source.ip, remoteShortcut._source.ip)) { diff = true; }
+          if (!compareArrays(localShortcut._source.string, remoteShortcut._source.string)) { diff = true; }
+          if (!compareArrays(localShortcut._source.number, remoteShortcut._source.number)) { diff = true; }
```

Comment:
style and slightly more efficient (stop after first diff) woudl be to do
let diff = A ||
               B ||
               C ||

although lint might get mad.